ScriptEngine/src/api/ServerAPI.cpp

```diff
@@ -37,3 +40,84 @@ Local<Value> McClass::setMaxNumPlayers(const Arguments& args) {
     }
     CATCH("Fail in setMaxPlayers!")
 }
+
+Local<Value> McClass::getTime(const Arguments& args) {
+    int option = 0; // option: 0: daytime, 1: gametime, 2: day
+
+    if(args.size() > 0) {
+        CHECK_ARG_TYPE(args[0], ValueKind::kNumber)
+        option = args[0].asNumber().toInt32();
+    }
+
+    if (option == 0)
+        return Number::newNumber(Global<Level>->getTime() % 24000);
+    else if (option == 1)
+        return Number::newNumber((int)Global<Level>->getCurrentTick());
+    else if (option == 2)
+        return Number::newNumber(Global<Level>->getTime() / 24000);
+    else
+        throw script::Exception("The range of this argument is between 0 and 2");
+
+}
+
+Local<Value> McClass::setTime(const Arguments& args) {
+    CHECK_ARGS_COUNT(args, 1)
+    CHECK_ARG_TYPE(args[0], ValueKind::kNumber)
+
+    try {
+        int currentTime = Global<Level>->getTime();
+        int targetTime = args[0].asNumber().toInt32(); // Tick
+
+        int newTime = currentTime;
+        int currentTimeOfDay = currentTime % 24000;
+
+        if ( targetTime > currentTime % 24000 )
+            newTime = currentTime + targetTime - currentTimeOfDay;
+        else if ( targetTime < currentTimeOfDay )
+            newTime = currentTime + targetTime + 24000 - currentTimeOfDay;
+
+        Global<Level>->setTime(newTime);
+        SetTimePacket pkt = SetTimePacket(newTime);
+        LoopbackPacketSender* pktSender = (LoopbackPacketSender*)Global<Level>->getPacketSender();
+        pktSender->send(pkt);
+    }
+    CATCH("Fail in setTime!")
+
+    return Boolean::newBoolean(true);
+}
+
+Local<Value> McClass::getWeather(const Arguments& args) { // weather: 0: Clear, 1: Rain, 2: Thunder
+    if (Global<Level>->getLevelData().isLightning())
+        return Number::newNumber(2);
+    else if (Global<Level>->getLevelData().isRaining())
+        return Number::newNumber(1);
+
+    return Number::newNumber(0);
+}
+
+Local<Value> McClass::setWeather(const Arguments& args) {
+    CHECK_ARGS_COUNT(args, 1)
+    CHECK_ARG_TYPE(args[0], ValueKind::kNumber)
+
+    int duration, weather = args[0].asNumber().toInt32(); // weather: 0: Clear, 1: Rain, 2: Thunder
```

Comment:
最好给duration赋个初值，这是局部变量
```suggestion
    int duration = 0;
    int weather = args[0].asNumber().toInt32(); // weather: 0: Clear, 1: Rain, 2: Thunder
```